### PR TITLE
Cleanup refactor

### PR DESCRIPTION
**Pull request by Neloop**

### CODE REVIEW

File: app/commands/cleanup/CleanupExerciseConfigs.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+  /** @var EntityRepository */
+  private $assignments;
+
+  /** @var EntityManager */
+  private $entityManager;
+
+  public function __construct(EntityManager $entityManager) {
+    parent::__construct();
+    $this->exercises = $entityManager->getRepository(Exercise::class); // even deleted exercises has to be found
```
Comment (Teyras) [inline app/commands/cleanup/CleanupExerciseConfigs.php]: Why don't you inject those classes directly (as constructor arguments)? `$em->getRepository()` doesn't guarantee you'll get an `EntityRepository` - you can configure the repository class in the `@ORM\Entity` annotation. Maybe adding methods that bypass the deletedAt filter would be a better solution.

File: readme.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ recommended way of ensuring this is using a crontab like this:
 
 ```
 04	00	*	*	*	php www/index.php notifications:assignment-deadlines "1 day"
-02	00	*	*	*	php www/index.php uploads:cleanup
+02	00	*	*	*	php www/index.php db:cleanup:uploads
```
Comment (Teyras) [inline readme.md]: You could probably add the other commands here too.

Comment (Neloop) [reply readme.md]: Well... it depends if we do want to do this regularly or not. In my opinion cleanups of configs and localized texts might by done once or twice a year, not every day or week...

File: app/commands/cleanup/CleanupExerciseConfigs.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    $deleteQuery->setParameter("ids", $usedLimits, Connection::PARAM_STR_ARRAY);
+    return $deleteQuery->execute();
+  }
+
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $now = new DateTime();
+    $limit = clone $now;
+    $limit->modify("-14 days");
+
+    $deletedEnv = $this->cleanupEnvironmentConfigs($limit);
```
[the remark this answers is not in the document]
Comment (Neloop) [reply app/commands/cleanup/CleanupExerciseConfigs.php]: Ok, can do...

File: app/commands/cleanup/CleanupExerciseConfigs.php (hunk 1 above)
Comment (Neloop) [reply app/commands/cleanup/CleanupExerciseConfigs.php]: Ok, I'll add new method to base soft delete repository

**Pull request merged**